build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/datastore/DatastoreAdapter.java

```diff
@@ -294,9 +294,12 @@ public void storeRejected(BuildTask buildTask) throws DatastoreException {
                 .user(buildTask.getUser())
                 .submitTime(buildTask.getSubmitTime())
                 .startTime(buildTask.getStartTime())
-                .productMilestone(buildTask.getProductMilestone())
                 .temporaryBuild(buildOptions.isTemporaryBuild());
 
+        if (! buildOptions.isTemporaryBuild()) {
```

Comment:
Not sure if this is the best place to put the logic. Shouldn't this be a concern of the Builder.build() implementation? I.e. if the temporary flag is true on the builder it will ignore the milestone value. That would ensure that temporary builds' records will always be consistent even when initialized anywhere else than in the DatastoreAdapter. OTOH not sure if the logic will ever be implemented elsewhere so not a big deal.

And why is the space after the exclamation mark? It is not supposed to be there! :-)

Reply:
Moved to the model and added also PrePersist check